src/python/grpcio/grpc/_cython/_cygrpc/aio/callback_common.pyx.pxi

```diff
@@ -70,7 +70,8 @@ cdef CallbackFailureHandler CQ_SHUTDOWN_FAILURE_HANDLER = CallbackFailureHandler
     InternalError)
 
 
-class ExecuteBatchError(Exception): pass
+class ExecuteBatchError(InternalError):
+    """Raised upon execute batch returns a fail from Core."""
```

Comment:
Nit:
s/upon/when/ ("upon" takes a noun phrase; "when" takes a verb phrase) 
s/fail/failure/ ("fail" is only used as a noun colloquially)

Reply:
Good catch!